pkg/cmd/cli/cmd/set/volume_test.go

```diff
@@ -0,0 +1,109 @@
+package set
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/testapi"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/client/unversioned/fake"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+func makeFakePod() *api.Pod {
+	fakePod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Namespace: "default",
+			Name:      "fakepod",
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name: "fake-container",
+					VolumeMounts: []api.VolumeMount{
+						{
+							Name:      "fake-mount",
+							MountPath: "/var/www/html",
+						},
+					},
+				},
+			},
+			Volumes: []api.Volume{
+				{
+					Name: "fake-mount",
+					VolumeSource: api.VolumeSource{
+						HostPath: &api.HostPathVolumeSource{
+							Path: "/var/www/html",
+						},
+					},
+				},
+			},
+		},
+	}
+	return fakePod
+}
+
+func getFakeMapping() *meta.RESTMapping {
+	fakeMapping := &meta.RESTMapping{
+		Resource: "fake-mount",
+		GroupVersionKind: unversioned.GroupVersionKind{
+			Group:   "test.group",
+			Version: "v1",
+		},
+		ObjectConvertor: api.Scheme,
+	}
+	return fakeMapping
+}
+
+func TestVolumeRunWithPatch(t *testing.T) {
```

Comment:
okay, will add. But partly It was intentional, the add path requires mocked objects which are rather complicated to fake/mock. :(